src/index.js

```diff
@@ -1,259 +1,75 @@
 /**
- * @copyright 2016, Prometheus Research, LLC
- * @copyright 2016-present, Andrey Popp <8mayday@gmail.com>
+ * @copyright 2017-present, Andrey Popp
+ *
+ * @flow
  */
 
-import invariant from 'invariant';
-import React from 'react';
-import {__Reactor as Reactor, __captureDereferences, struct, isDerivable} from 'derivable';
-
-/**
- * Reactor which wraps React component and schedules its update when
- * dependencies change.
- */
-export class ReactUpdateReactor extends Reactor {
-  constructor(component) {
-    super();
-    this.react = this.constructor.prototype.react;
-    this.component = component;
-    this.dependencies = null;
-  }
-
-  setDependenciesFrom(thunk) {
-    let result;
-    this.setDependencies(__captureDereferences(() => (result = thunk())));
-    return result;
-  }
-
-  setDependencies(dependencies) {
-    this._parent = struct(dependencies);
-    this.dependencies = dependencies;
-  }
+import * as React from 'react';
+import * as Derivable from 'derivable';
 
-  react() {
-    this.stop();
-    this.component.forceUpdate();
-  }
-
-  shutdown() {
-    this.stop();
-    this.dependencies = null;
-    this.component = null;
-  }
+export function renderMap<V>(d: Derivable.Derivable<V>, f: V => React.Node): React.Node {
+  return <Reactive d={d} f={f} />;
 }
 
-/**
- * Produce reactive component out of original React component.
- *
- * Reactive component re-render when one of the reactive values used in render()
- * funciton changes.
- *
- * @example
- *
- * let Hello = props =>
- *   <div>{props.message.get()}</div>
- *
- * let ReactiveHello = reactive(Hello)
- *
- * @example
- *
- * class Hello extends React.Component {
- *   render() {
- *     return <div>{props.message.get()}</div>
- *   }
- * }
- *
- * let ReactiveHello = reactive(Hello)
- */
-export function reactive(Component) {
-  return decorateWith(Component, makeReactiveComponent);
-}
-
-/**
- * Produce pure reactive component out of original React component.
- *
- * Pure reactive component behave the same as reactive component but have
- * additional optimization through shouldComponentUpdate which prevents
- * re-rendering if props / state aren't changed.
- *
- * @example
- *
- * let Hello = props =>
- *   <div>{props.message.get()}</div>
- *
- * let PureHello = pure(Hello)
- *
- * @example
- *
- * class Hello extends React.Component {
- *   render() {
- *     return <div>{props.message.get()}</div>
- *   }
- * }
- *
- * let PureHello = pure(Hello)
- */
-export function pure(Component) {
-  Component = decorateWith(Component, makeReactiveComponent);
-  return decorateWith(Component, makePureComponent);
+export function render<V: React.Node>(d: Derivable.Derivable<V>): React.Node {
+  return <Reactive d={d} f={renderValue} />;
```

Comment:
Somebody would like to use this component directly :)

Reply:
Why?

Reply:
I know those guys from new context rfc

Reply:
Can you elaborate?

Reply:
Some guys from context rfc are jealously prefer jsx over function calls and hocs.

Reply:
Does it matter? I think we'd like to settle on what React Context API officially settles.

Reply:
This style seems to me less noisy than using JSX with function-as-children.

Reply:
I don't force. Just a note. I'm not a fan of pure jsx too.